flink-formats/flink-avro/src/main/java/org/apache/flink/formats/avro/typeutils/AvroSchemaConverter.java

```diff
@@ -297,32 +300,53 @@ private static DataType convertToDataType(Schema schema) {
 	 * @return Avro's {@link Schema} matching this logical type.
 	 */
 	public static Schema convertToSchema(LogicalType logicalType) {
+		// If it is parsing the root row type, switches from nullable true to false
+		// because a nullable row type is meaningless and would generate wrong schema.
+		if (logicalType.getTypeRoot() == LogicalTypeRoot.ROW
```

Comment:
I think we can make the parameter to be `RowType`, that would be make sense to use it as the top-level row type and not generate nullable for it. Besides, would be better to add comments in the Javadoc. Currently, this method has the same Javadoc with `convertToSchema(LogicalType logicalType, String rowName)`.

Reply:
Although the `AvroSchemaConverter` is a tool class, it is still used as a public API, so i'm inclined to keep the signature unchanged. Another reason is that only logical type is enough for the conversion.

Have added more documents to the methods.